modules/ding_user_frontend/ding_user_frontend.module

```diff
@@ -330,3 +329,54 @@ function ding_user_frontend_build_menu($path = NULL) {
 
   return $menu;
 }
+
+/**
+ * Implements hook_outbound_alter().
+ *
+ * If URLs like user/1234/view are cached in e.g. varnish, then other users
+ * may be exposed to these URLs.
+ *
+ * This function normalizes the user/* paths, by replacing the uid with "me".
+ */
+function ding_user_frontend_url_outbound_alter(&$path, &$options, $original_path) {
+  global $user;
+  if (preg_match('/^user\/(\d+)\/(.*)$/', $path, $matches)) {
+    // As default change the path to use "me".
+    $path = 'user/me/' . $matches[2];
+
+    // To allow site administrators to edit other user they have to be filter
+    // out of this alter. When accessing another drupal user account.
+    if ($user->uid != $matches[1] && (array_search("administrators", $user->roles) || array_search("local administrator", $user->roles))) {
+      // This is an "administrators" or "local administrator" trying to edit
+      // another user. To "allow" this we reset the path.
+      $path = $original_path;
```

Comment:
Is all this necessary at all?

I just don't see why you would generate links on the form /user/me/* for other users than the current user. IMO the only check needed is to verify that the link is for the current user ($user->uid == $matches[0]).